File: src/p1xxx/p1518.py
from leetcode_prelude import *


# Problem 1518
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        count = numBottles
        empty = numBottles

        while empty >= numExchange:
            q, r = divmod(empty, numExchange)
            count += q
            empty = r + q

        return count

File: src/p1xxx/p1518.rs
// Problem 1518
impl Solution {
    pub fn num_water_bottles(mut num_bottles: i32, mut num_exchange: i32) -> i32 {
        let mut done = 0;
        let mut empty_bottles = 0;

        while num_bottles > 0 {
            done += num_bottles;

            empty_bottles += num_bottles;

            let (q, r) = (empty_bottles / num_exchange, empty_bottles % num_exchange);
            empty_bottles = r;
            num_bottles = q;
        }

        done
    }
}
